src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -0,0 +1,1262 @@
+    {
+        const auto oldIsQuakeMode = IsQuakeWindow();
+
+        const auto changed = _WindowName != name;
+        if (changed)
+        {
+            _WindowName = name;
+            if (_root)
+            {
+                _root->WindowNameChanged();
+
+                // If we're entering quake mode, or leaving it
+                if (IsQuakeWindow() != oldIsQuakeMode)
+                {
+                    // If we're entering Quake Mode from ~Focus Mode, then this will enter Focus Mode
+                    // If we're entering Quake Mode from Focus Mode, then this will do nothing
+                    // If we're leaving Quake Mode (we're already in Focus Mode), then this will do nothing
+                    _root->SetFocusMode(true);
+                    _IsQuakeWindowChangedHandlers(*this, nullptr);
+                }
+            }
+        }
+    }
+    uint64_t TerminalWindow::WindowId() const noexcept
+    {
+        return _WindowId;
+    }
+    void TerminalWindow::WindowId(const uint64_t& id)
+    {
+        if (_WindowId != id)
+        {
+            _WindowId = id;
+            if (_root)
+            {
+                _root->WindowNameChanged();
+            }
+        }
+    }
+
+    // Method Description:
+    // - Returns a label like "Window: 1234" for the ID of this window
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - a string for displaying the name of the window.
+    winrt::hstring TerminalWindow::WindowIdForDisplay() const noexcept
+    {
+        return winrt::hstring{ fmt::format(L"{}: {}",
+                                           std::wstring_view(RS_(L"WindowIdLabel")),
+                                           _WindowId) };
+    }
+
+    // Method Description:
+    // - Returns a label like "<unnamed window>" when the window has no name, or the name of the window.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - a string for displaying the name of the window.
+    winrt::hstring TerminalWindow::WindowNameForDisplay() const noexcept
+    {
+        return _WindowName.empty() ?
+                   winrt::hstring{ fmt::format(L"<{}>", RS_(L"UnnamedWindowName")) } :
+                   _WindowName;
+    }
+    bool TerminalWindow::IsQuakeWindow() const noexcept
+    {
+        return _WindowName == QuakeWindowName;
+    }
+
+    void TerminalWindow::RequestExitFullscreen()
+    {
+        _root->SetFullscreen(false);
+    }
+
+    bool TerminalWindow::AutoHideWindow()
+    {
+        return _settings.GlobalSettings().AutoHideWindow();
+    }
+
+    void TerminalWindow::UpdateSettingsHandler(const winrt::IInspectable& /*sender*/,
+                                               const winrt::TerminalApp::SettingsLoadEventArgs& arg)
+    {
+        UpdateSettings(arg);
+    }
+
+    bool TerminalWindow::ShouldImmediatelyHandoffToElevated()
+    {
+        return _root != nullptr ? _root->ShouldImmediatelyHandoffToElevated(_settings) : false;

```

Comment:
this also doesn't seem like something the UI should be able to answer

Reply:
resolving as a part of the "commandline args are per-window" conversation